### PR TITLE
Document custom session parameters via env variables

### PR DESCRIPTION
**Pull request by lucas-koehler**

### CODE REVIEW

File: content/documentation/sessionParameters/content.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
++++
+
+This tutorial describes how custom parameters are injected into session pods as environment variables.
+This may be used to inject parameters into a session that are not equal for all sessions of an app definition.
+For instance, access tokens or dynamic parameters such as a Git repository to check out.
+
+## Adding environment variables via REST API
+
+The simplest way to start a Session with custom environment variables is using Theia Cloud's REST API.
+Theia Cloud offers npm package [@eclipse-theiacloud/common](https://www.npmjs.com/package/@eclipse-theiacloud/common) to access the REST API conveniently.
```
Comment (jfaltermeier) [inline content/documentation/sessionParameters/content.md:18]:
```suggestion
Theia Cloud offers an npm package [@eclipse-theiacloud/common](https://www.npmjs.com/package/@eclipse-theiacloud/common) to access the REST API conveniently.
```

File: content/documentation/sessionParameters/content.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+It offers to provide environment variables directly or as lists of [config maps](https://kubernetes.io/docs/concepts/configuration/configmap/) or [secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to read in.
+
+The following code snippet show how to add environment variables while starting a session.
```
Comment (jfaltermeier) [inline content/documentation/sessionParameters/content.md:21]:
```suggestion
The following code snippet shows how to add environment variables while starting a session.
```

File: content/documentation/sessionParameters/content.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+Thereby, the whole `env` as well as its three properties `fromMap`, `fromConfigMaps` and `fromSecrets` are optional.
+The snippet assumes that the parameters `accessToken` and `user` are present as variables based on the authenticated user.
+You can have a look at the [Try Now Page](https://github.com/eclipsesource/theia-cloud/blob/main/node/try-now-page/src/App.tsx) to see an example of how these are derived when logging in via Keycloak.
+
+```typescript
+import { getTheiaCloudConfig, SessionStartRequest, TheiaCloud } from '@eclipse-theiacloud/common';
+
+// Get common parameters from Theia Cloud configuration
+const { appDefinition, appId, serviceUrl} = getTheiaCloudConfig();
```
Comment (jfaltermeier) [inline content/documentation/sessionParameters/content.md:30]:
```suggestion
const { appDefinition, appId, serviceUrl } = getTheiaCloudConfig();
```

File: content/documentation/sessionParameters/content.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+// Send request
+TheiaCloud.Session.startSession(request);
+```
+
+## Adding environment variables via Session CR
+
+The environment variables can also be configured directly through the Session custom resource by adding one or multiple of the following properties:
+
+- [envVars](https://github.com/eclipsesource/theia-cloud-helm/blob/4cd9d98d30cebe8d31e7084369878c1c2d28776c/charts/theia-cloud-crds/templates/session-spec-resource.yaml#L46-L49): This property is a map that allows you to define environment variables directly. Each entry in the map consists of the environment variable name and its value. This method is suitable for standard configuration needs and facilitates individual configuration for each session.
```
Comment (jfaltermeier) [inline content/documentation/sessionParameters/content.md:57]: I don't like the links to specific commits, however I don't see a better option. 
Maybe we should keep track where such links are used, so we may check them regularly?

File: content/documentation/sessionParameters/content.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  serviceUrl,
+  accessToken,
+  appId,
+  user,
+  appDefinition,
+  env: {
+    fromMap: {
+      mykey: "myvalue"
+    },
+    fromConfigMaps: [ "session-config-map-1", "config-map-2"],
```
Comment (jfaltermeier) [inline content/documentation/sessionParameters/content.md:44]:
```suggestion
    fromConfigMaps: [ "session-config-map-1", "config-map-2" ],
```

File: content/documentation/sessionParameters/content.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    fromSecrets: [ "session-secret-1", "session-secret-2"]
```
Comment (jfaltermeier) [inline content/documentation/sessionParameters/content.md:45]:
```suggestion
    fromSecrets: [ "session-secret-1", "session-secret-2" ]
```

**Review by jfaltermeier — COMMENTED** [review]
Already looks very good to me, I have a few smaller suggestions though.

File: content/documentation/sessionParameters/content.md (hunk 4 above)
Comment (lucas-koehler) [reply content/documentation/sessionParameters/content.md:57]: Mhm, yes I agree that it is not optimal to use it like that. If we point to main, we would also need to keep track and check links regularly. Especially with references to CRDs where a new version will throw off the referenced lines completely.

Maybe we could keep a file like `repository-links` in the root of the repository that either keeps entries for every link and its file or just files that contain repository links. What do you think?

File: content/documentation/sessionParameters/content.md (hunk 1 above)
Comment (jfaltermeier) [reply content/documentation/sessionParameters/content.md:18]: Sorry, I read the markdown formatting as actual brackets :smile:

File: content/documentation/sessionParameters/content.md (hunk 4 above)
Comment (jfaltermeier) [reply content/documentation/sessionParameters/content.md:57]: Sounds good, I think keeping the files should be enough. 
I'll add it to the release template to check for outdated links after a release

**Comment by lucas-koehler** [comment]
@jfaltermeier I added the `repository-links` file and added the content.md for the session parameters. For now, I left in the link to a specific commit because it makes it easier to see what should be referenced even if the file changes on main.

**Review by jfaltermeier — APPROVED** [review]

**Pull request merged**